src/bulkDataRequestCache.ts

```diff
@@ -64,61 +64,33 @@ export abstract class BulkDataRequestCache extends TTLConfig<TTLConfig.Options,
     Logger.childFromRoot('DataRequestCache').debug(`bulk cache saved for ${bulkRequestId}`);
   }
 
-  public async resolveResumeOptionsFromCache(
-    bulkJobId: string | undefined,
-    useMostRecent: boolean,
-    org: Org | undefined,
-    apiVersion: string | undefined
-  ): Promise<ResumeOptions> {
-    if (!useMostRecent && !bulkJobId) {
-      throw messages.createError('bulkRequestIdRequiredWhenNotUsingMostRecent');
-    }
-    const resumeOptions = {
-      operation: 'query',
-      query: '',
-      pollingOptions: { pollTimeout: 0, pollInterval: 0 },
-    } satisfies Pick<ResumeOptions['options'], 'operation' | 'query' | 'pollingOptions'>;
-
-    if (useMostRecent) {
+  public async resolveResumeOptionsFromCache(jobIdOrMostRecent: string | boolean): Promise<ResumeBulkImportOptions> {
+    if (typeof jobIdOrMostRecent === 'boolean') {
       const key = this.getLatestKey();
-      if (key) {
-        // key definitely exists because it came from the cache
-        const entry = this.get(key);
-
-        return {
-          jobInfo: { id: entry.jobId },
-          options: {
-            ...resumeOptions,
-            connection: (await Org.create({ aliasOrUsername: entry.username })).getConnection(apiVersion),
-          },
-        };
-      }
-    }
-    if (bulkJobId) {
-      const entry = this.get(bulkJobId);
-      if (entry) {
-        return {
-          jobInfo: { id: entry.jobId },
-          options: {
-            ...resumeOptions,
-            connection: (await Org.create({ aliasOrUsername: entry.username })).getConnection(apiVersion),
-          },
-        };
-      } else if (org) {
-        return {
-          jobInfo: { id: bulkJobId },
-          options: {
-            ...resumeOptions,
-            connection: org.getConnection(apiVersion),
-          },
-        };
-      } else {
-        throw messages.createError('cannotCreateResumeOptionsWithoutAnOrg');
+      if (!key) {
+        throw messages.createError('error.missingCacheEntryError');
       }
-    } else if (useMostRecent) {
-      throw messages.createError('error.missingCacheEntryError');
+      // key definitely exists because it came from the cache
+      const entry = this.get(key);
+
+      return {
+        jobInfo: { id: entry.jobId },
+        options: {
+          connection: (await Org.create({ aliasOrUsername: entry.username })).getConnection(),
+        },
+      };
     } else {
-      throw messages.createError('bulkRequestIdRequiredWhenNotUsingMostRecent');
+      const entry = this.get(jobIdOrMostRecent);
+      if (!entry) {
+        throw messages.createError('error.bulkRequestIdNotFound', [jobIdOrMostRecent]);
+      }
+
+      return {
+        jobInfo: { id: entry.jobId },
+        options: {
+          connection: (await Org.create({ aliasOrUsername: entry.username })).getConnection(),
+        },
+      };
```

Comment:
simplified `resolveResumeOptionsFromCache` so that it matches `data import/export`'s cache resolver.
The previous implementation had a few unused properties and one (undocumented?) functionality:
passing a job ID that didn't exist in the cache wouldn't cause an error, the resolver would return it as if it was found in the local cache.
Added code ƒor specific bulk commands using this to avoid breaking changes.